packages/util-retry/src/StandardRetryStrategy.ts

```diff
@@ -1,47 +1,88 @@
 import { Provider, RetryErrorInfo, RetryErrorType, RetryStrategyV2, StandardRetryToken } from "@aws-sdk/types";
 
 import { DEFAULT_MAX_ATTEMPTS, RETRY_MODES } from "./config";
-import { DEFAULT_RETRY_DELAY_BASE, INITIAL_RETRY_TOKENS } from "./constants";
-import { getDefaultRetryToken } from "./defaultRetryToken";
+import {
+  DEFAULT_RETRY_DELAY_BASE,
+  INITIAL_RETRY_TOKENS,
+  NO_RETRY_INCREMENT,
+  RETRY_COST,
+  THROTTLING_RETRY_DELAY_BASE,
+  TIMEOUT_RETRY_COST,
+} from "./constants";
+import { getDefaultRetryBackoffStrategy } from "./defaultRetryBackoffStrategy";
+import { createDefaultRetryToken } from "./defaultRetryToken";
 
 /**
  * @public
  */
 export class StandardRetryStrategy implements RetryStrategyV2 {
   public readonly mode: string = RETRY_MODES.STANDARD;
-  private retryToken: StandardRetryToken;
+  private capacity: number = INITIAL_RETRY_TOKENS;
+  private readonly retryBackoffStrategy = getDefaultRetryBackoffStrategy();
   private readonly maxAttemptsProvider: Provider<number>;
 
   constructor(maxAttempts: number);
   constructor(maxAttemptsProvider: Provider<number>);
   constructor(private readonly maxAttempts: number | Provider<number>) {
-    this.retryToken = getDefaultRetryToken(INITIAL_RETRY_TOKENS, DEFAULT_RETRY_DELAY_BASE);
     this.maxAttemptsProvider = typeof maxAttempts === "function" ? maxAttempts : async () => maxAttempts;
   }
 
   public async acquireInitialRetryToken(retryTokenScope: string): Promise<StandardRetryToken> {
-    return this.retryToken;
+    return createDefaultRetryToken({
+      availableCapacity: this.capacity,
+      retryDelay: DEFAULT_RETRY_DELAY_BASE,
+      retryCount: 0,
+    });
   }
 
   public async refreshRetryTokenForRetry(
-    tokenToRenew: StandardRetryToken,
+    token: StandardRetryToken,
     errorInfo: RetryErrorInfo
   ): Promise<StandardRetryToken> {
     const maxAttempts = await this.getMaxAttempts();
 
-    if (this.shouldRetry(tokenToRenew, errorInfo, maxAttempts)) {
-      tokenToRenew.getRetryTokenCount(errorInfo);
-      return tokenToRenew;
+    const getCapacityCost = (errorType: RetryErrorType) =>
+      errorType === "TRANSIENT" ? TIMEOUT_RETRY_COST : RETRY_COST;
+
+    if (this.shouldRetry(token, errorInfo, maxAttempts)) {
+      const errorType = errorInfo.errorType;
+      const capacityCost = getCapacityCost(errorType);
+      const delayBase = errorType === "THROTTLING" ? THROTTLING_RETRY_DELAY_BASE : DEFAULT_RETRY_DELAY_BASE;
+      this.retryBackoffStrategy.setDelayBase(delayBase);
```

Comment:
nit: delayBase variable can be removed as it's used only once
```suggestion

      this.retryBackoffStrategy.setDelayBase(
        errorType === "THROTTLING" ? THROTTLING_RETRY_DELAY_BASE : DEFAULT_RETRY_DELAY_BASE
      );
```